public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -178,40 +184,67 @@ export class GraphiteDatasource
   }
 
   query(options: DataQueryRequest<GraphiteQuery>): Observable<DataQueryResponse> {
-    const graphOptions = {
-      from: this.translateTime(options.range.from, false, options.timezone),
-      until: this.translateTime(options.range.to, true, options.timezone),
-      targets: options.targets,
-      format: (options as any).format,
-      cacheTimeout: options.cacheTimeout || this.cacheTimeout,
-      maxDataPoints: options.maxDataPoints,
-    };
+    const streams: Array<Observable<DataQueryResponse>> = [];
 
-    const params = this.buildGraphiteParams(graphOptions, options.scopedVars);
-    if (params.length === 0) {
-      return of({ data: [] });
-    }
+    for (const target of options.targets) {
+      if (target.hide) {
+        continue;
```

Comment:
I think we can flip the if and we can remove the `continue`? WDYT?

Reply:
For hiding the query, I used Sriram's example from DataDog, but on looking further into Graphite, the "hide" feature is handled elsewhere, in ```buildGraphiteParams```. I think the ``` if (target.hide)``` condition is extra and we can remove it. Thank you!

Reply:
This may be a case for refactoring in the future though, moving the hide feature into the query function.